kit/src/components/message/mod.rs

```diff
@@ -489,102 +524,66 @@ fn markdown(text: &str, emojis: bool) -> String {
-                    let txt: pulldown_cmark::CowStr<'_> = if in_paragraph {
-                        text.replace("\n\n", "<br/>").into()
-                    } else {
-                        text.into()
-                    };
+            }
+            pulldown_cmark::Event::Start(Tag::Paragraph) => {
+                in_paragraph = true;
+                html_output.push_str("<p>");
+            }
+            pulldown_cmark::Event::End(Tag::Paragraph) => {
+                in_paragraph = false;
+            }
+            pulldown_cmark::Event::Text(t) => {
+                let text = if emojis || in_code_block {
+                    process_string(&t, |s| stack_processor(s, in_code_block, emojis))
+                } else {
+                    t.to_string()
+                };
+                let txt: pulldown_cmark::CowStr<'_> = if in_paragraph {
+                    text.replace("\n\n", "<br/>").into()
+                } else {
+                    text.into()
+                };
+                if in_code_block {
+                    html_output.push_str(&txt);
+                } else {
                     pulldown_cmark::html::push_html(
                         &mut html_output,
                         std::iter::once(pulldown_cmark::Event::Text(txt)),
                     );
                 }
-                pulldown_cmark::Event::Start(pulldown_cmark::Tag::CodeBlock(code_block_kind)) => {
-                    add_text_language = false;
+            }
+            event => {
+                if let pulldown_cmark::Event::Start(pulldown_cmark::Tag::CodeBlock(_)) = event {
                     in_code_block = true;
-                    match code_block_kind {
-                        CodeBlockKind::Fenced(language) => {
-                            let language = if language.is_empty() {
-                                "text"
-                            } else {
-                                &language
-                            };
-
-                            html_output
-                                .push_str(&format!("<pre><code class=\"language-{}\">", language))
-                        }
-                        _ => html_output.push_str("<pre><code class=\"language-text\">"),
-                    }
-                }
-                pulldown_cmark::Event::End(pulldown_cmark::Tag::CodeBlock(_)) => {
-                    if in_code_block && line_trim == "```" {
-                        in_code_block = false;
-                        add_text_language = true;
-                        // HACK: To close block code is necessary to push tags 2 times
-                        html_output.push_str("</code></pre>");
-                        html_output.push_str("</code></pre>");
-                    }
+                } else if let pulldown_cmark::Event::End(pulldown_cmark::Tag::CodeBlock(_)) = event
+                {
+                    in_code_block = false;
                 }
```

Comment:
might be more idiomatic to use a match condition. Something like

```rust
match event {
    pulldown_cmark::Event::Start(pulldown_cmark::Tag::CodeBlock(_)) => in_code_block = true,
    pulldown_cmark::Event::End(pulldown_cmark::Tag::CodeBlock(_)) => in_code_block = false,
    _ => {}
};
```

Reply:
even for only 2 cases? hmm ok